Resolve imports without extension

pago opened the pull request:
In my use case, I have a project where imports are written without a file extension, i.e.

```js
import { foo } from './foo'; // => foo.js or foo/index.js
```

This PR uses `require.resolve` to figure out the actual file name if no extension was specified.
Most probably, a more advanced resolution would be better but this should be good enough and avoids some confusion.

Additionally, I have fixed a few problems I ran into.

pago commented on lib/extractors/exported.js:
```diff
@@ -28,7 +28,8 @@ function walkExported(ast, data, addComment) {
     if (!hasJSDocComment(path)) {
-      return [addBlankComment(data, path, path.node)];
+      var added = addBlankComment(data, path, path.node);
+      return added ? [added] : [];
```
This avoids running into a NPE later on when `addBlankComment` does not return a value.

arv commented on lib/extractors/exported.js:
```diff
@@ -136,7 +137,12 @@ function traverseExportedSubtree(path, data, addComments, overrideName) {
+function getCachedData(dataCache, filePath) {
+  var path = filePath;
+  if (!nodePath.extname(path)) {
+    path = require.resolve(path);
```
I think this makes sense. A lot of people use node's resolve mechanism today.

arv commented on lib/parsers/javascript.js:
```diff
@@ -70,8 +70,13 @@ function _addComment(visited, data, commentValue, commentLoc, path, nodeLoc, inc
-        context.code = data.source.substring
-          .apply(data.source, path.parentPath.node.range);
+        var parentNode = path.parentPath.node;
+        if (parentNode.range) {
```
We updated our babel deps pretty recently. I think we should be able to remove this.

pago commented:
@arv I've added test coverage and removed the fallback for the old Babel API.
Test is dead simple (just removed `.js` from the test fixture in one occurance) but I guess its ok.
